sockeye/utils.py

```diff
@@ -615,3 +615,53 @@ def all_gather_object(obj: T) -> List[T]:
     obj_list = [None] * torch.distributed.get_world_size()  # type: List[T]
     torch.distributed.all_gather_object(obj_list, obj)
     return obj_list
+
+
+def count_seq_len(sample: str, count_type: str = 'char', replace_tokens: List = []) -> int:
+    """
+    Count sequence length, after replacing word segmentation token/s.
+    """
+    if len(replace_tokens) >= 1:
+        for r in replace_tokens:
+            sample = sample.replace(r, '')
+    if count_type == 'char':
+        return len(sample.replace(' ', ''))
+    else:
+        return len(sample.split(' '))
+
+
+def compute_isometric_score(hypothesis: str, hypothesis_score: float, source: str,
+                            isometric_metric: str = 'isometric-ratio', isometric_alpha: float = 0.5) -> float:
+    """
+    Compute hypothesis to source isometric score using sample char length
+    and isometric metric (ratio, diff, lc).
+        - isometric-diff: https://aclanthology.org/W19-5210.pdf
+        - isometric-ratio/lc: https://arxiv.org/pdf/2110.03847.pdf
+    :return isometric score
+    """
+    count_type = 'char'  # for isometric scoring, count length in characters
+    replace_tokens = C.TOKEN_SEGMENTATION_MARKERS
```

Comment:
if this is where you hardcode the segmentation markers, why not make the replace_tokens argument in `count_seq_len` default directly to C.TOKEN_SEGMENTATION_MARKERS?

Reply:
this will make `count_seq_len` less flexible, if called from another func to handle different tokens for replacement. i added Sockeye wide constants `SEQ_LEN_IN_*` to avoid hard coding `char`, and assuming common sequence length counting types (char, tokens, and words).